Automattic/wp-calypso
# Jetpack Error UX: Add component that checks Jetpack connection health

Related to https://github.com/Automattic/dotcom-forge/issues/3202

## Proposed Changes

In this diff, I propose adding a new component that checks Jetpack connection health and displays an accurate message that can let the user fix the connection.

The component should only execute a request to the API endpoint if we suspect the connection can be broken. The connection can be broken if another request made to an external site via the proxy failed.

## Testing Instructions



1. Apply patch D116410-code
2. Load the posts page for a blog with a broken connection
3. Confirm that the message is displayed
4. Confirm that there is a track logged for notice view and notice action click

![Screenshot 2023-08-02 at 10 26 22](https://github.com/Automattic/wp-calypso/assets/727413/d2939449-b4d1-487a-8b15-964dff1b5f59)

## Pre-merge Checklist
